Helper/Generator.php

```diff
@@ -476,17 +468,17 @@ protected function addOptionsToRecord($product) {
                 $values = $option->getValues();
                 if($values) {
                     foreach($values as $value) {
-                        $this->setRecordValue($field, $value->getTitle());
+                        Generator::setRecordValue($productRecord,$field, $value->getTitle(),$this->ignoreFields);
                     }
                 }
             }
         }
     }
 
-    protected function addImagesToRecord($product) {
-        $this->setRecordValue('cached_thumbnail', $this->getThumbnail($product));
+    protected function addImagesToRecord($product, &$productRecord) {
```

Comment:
what's this `&$` syntax

Reply:
you're passing the product by reference. so the value going in can change